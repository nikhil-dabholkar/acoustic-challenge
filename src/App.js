import React from 'react';
import './App.css';
import 'bootstrap/dist/css/bootstrap.min.css';
import 'ag-grid-community/dist/styles/ag-grid.css';
import 'ag-grid-community/dist/styles/ag-theme-alpine.css';
import  ArticleContainer  from "./containers/ArticleContainer";

function App() {
  return (
    <div className="App">
      <ArticleContainer></ArticleContainer>
    </div>
  );
}

export default App;
