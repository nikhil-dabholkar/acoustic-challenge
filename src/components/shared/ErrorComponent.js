import React from 'react';
import PropTypes from 'prop-types';
import { injectIntl, FormattedMessage } from 'react-intl';
import { InlineNotification } from 'carbon-components-react';
import styles from "./styles/ErrorComponent.module.scss";

function ErrorComponent({ error }) {
    if (error.response.data.errors.code && error.response.data.errors.message) {
        return (<div className={styles.container}><InlineNotification hideCloseButton kind="error" subtitle={<span>{error.response.data.errors.message}</span>} title={`${error.response.data.errors.code}: `}
        /></div>)
    } else {
        return (<div className={styles.container}><InlineNotification hideCloseButton
            kind="error"
            subtitle={<span><FormattedMessage id="errorComponent.default" defaultMessage="Something went wrong, Please try again after sometime."></FormattedMessage></span>}
            title="9008: "
        /></div>)
    }
}

ErrorComponent.propTypes = {
    intl: PropTypes.object.isRequired
}

export default injectIntl(React.memo(ErrorComponent));
