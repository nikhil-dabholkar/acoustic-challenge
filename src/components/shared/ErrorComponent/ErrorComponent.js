import React from 'react'
import PropTypes from 'prop-types'
import { injectIntl, FormattedMessage } from 'react-intl'
import { InlineNotification } from 'carbon-components-react'
import styles from './ErrorComponent.module.scss'

function ErrorComponent ({ error }) {
  if (error && error.response && error.response.data && error.response.data.errors && error.response.data.errors.code && error.response.data.errors.message) {
    return (<React.Fragment>
      <div data-testid="errorComponent" className={styles.container}><InlineNotification hideCloseButton kind="error" subtitle={<span>{error.response.data.errors.message}</span>} title={`${error.response.data.errors.code}: `}
      /></div>
    </React.Fragment>)
  } else {
    return (<React.Fragment>
      <div data-testid="errorComponent" className={styles.container}><InlineNotification hideCloseButton
        kind="error"
        subtitle={<span><FormattedMessage id="errorComponent.default" defaultMessage="Something went wrong, Please try again after sometime."></FormattedMessage></span>}
        title="9008: "
      /></div>
    </React.Fragment>)
  }
}

ErrorComponent.propTypes = {
  intl: PropTypes.object.isRequired,
  error: PropTypes.object.isRequired
}

export default injectIntl(React.memo(ErrorComponent))
