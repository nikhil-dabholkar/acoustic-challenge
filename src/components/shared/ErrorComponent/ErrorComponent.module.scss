.container {
    &>div {
        margin: 10% auto !important;
    }
}