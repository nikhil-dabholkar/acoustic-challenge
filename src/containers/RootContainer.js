import React from 'react';
import { BrowserRouter as Router, Route, Switch, Redirect, Link } from "react-router-dom";
import { lazy, Suspense } from 'react';
import { LoadingComponent } from '../components/index';
import ArticleContainer from './ArticleContainer';
import HomeContainer from './HomeContainer';
import HeaderContainer from './HeaderContainer';
import Jumbotron from 'react-bootstrap/Jumbotron'

const ErrorComponent = lazy(() => import("../components/shared/ErrorComponent"));

function RootContainer() {

    return (<React.Fragment>
        <Router>
            <HeaderContainer></HeaderContainer>
            <Jumbotron fluid>
                <Switch>
                    <Route path="/home" render={(props) => {
                        return (<Suspense fallback={<LoadingComponent />}>
                            <HomeContainer></HomeContainer>
                        </Suspense>);
                    }} exact />
                    <Route path="/articles" render={(props) => {
                        return (<Suspense fallback={<LoadingComponent />}>
                            <ArticleContainer></ArticleContainer>
                        </Suspense>);
                    }} />
                    <Route path="/" render={() => <Redirect to="/home" />} exact></Route>
                    <Route path="*" render={() => {
                        return (<Suspense fallback={<LoadingComponent />}>
                            <ErrorComponent />
                        </Suspense>)
                    }} exact />
                </Switch>
            </Jumbotron>
        </Router>
    </React.Fragment>
    );
}

export default RootContainer;
