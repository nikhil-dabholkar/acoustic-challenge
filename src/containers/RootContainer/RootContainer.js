import React from 'react';
import { BrowserRouter as Router, Route, Switch, Redirect, Link } from "react-router-dom";
import { lazy, Suspense } from 'react';
import PropTypes from 'prop-types';
import { injectIntl, FormattedDate, FormattedMessage } from 'react-intl';
import { LoadingComponent } from '../../components/index';
import ArticleContainer from '../ArticleContainer/ArticleContainer';
import HomeContainer from '../HomeContainer/HomeContainer';
import HeaderContainer from '../HeaderContainer/HeaderContainer';
import Jumbotron from 'react-bootstrap/Jumbotron';
import { ErrorComponent } from "../../components/index";

const HomeComponent = lazy(() => import("../../components/home/HomeComponent/HomeComponent"));

// const ErrorComponent = lazy(() => import("../../components/shared/ErrorComponent/ErrorComponent"));

function RootContainer({ intl }) {

    return (<React.Fragment>
        <div data-testid="rootContainer">
            <Router>
                <HeaderContainer></HeaderContainer>
                <Jumbotron fluid>
                    <Switch>

                        <Route path="/home" render={(props) => {
                            return (<Suspense fallback={<LoadingComponent />}>
                                <HomeContainer>
                                    <Suspense fallback={<LoadingComponent />}>
                                        <HomeComponent></HomeComponent>
                                    </Suspense>
                                </HomeContainer>
                            </Suspense>);
                        }} exact />
                        <Route path="/articles" render={(props) => {
                            return (<Suspense fallback={<LoadingComponent />}>
                                <ArticleContainer></ArticleContainer>
                            </Suspense>);
                        }} />
                        <Route path="/404" render={() => {
                            const errorObj = {
                                response: {
                                    data: {
                                        errors: {
                                            code: 56009,
                                            message: intl.formatMessage({ id: "rootContainer.invalidURL" })
                                        }
                                    }
                                }
                            }
                            return (
                                <ErrorComponent error={errorObj} />
                            );
                        }} />
                        <Route path="/" render={() => <Redirect to="/home" />} exact></Route>
                        <Route path="*" render={() => <Redirect to="/404" />}></Route>

                    </Switch>
                </Jumbotron>
            </Router>
        </div>
    </React.Fragment>
    );
}

RootContainer.propTypes = {
    intl: PropTypes.object.isRequired
}

export default injectIntl(RootContainer);
