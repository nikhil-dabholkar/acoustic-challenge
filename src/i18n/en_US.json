{
    "app.header": "English",
    "detailsComponent.dateTooltip": "Date and time is used when ordering a list of design articles. The time will not display in the website.",
    "detailsComponent.bodySectionTooltip": "The design topic, when displayed in the website, will interleave the text chunks and images to create a fluid layout.",
    "detailsComponent.topicTitle": "Topic title",
    "detailsComponent.authorName": "Author name",
    "detailsComponent.date": "Date",
    "detailsComponent.leadImage": "Lead image",
    "detailsComponent.leadImageCaption": "Lead image caption",
    "detailsComponent.LeadImageCredit": "Lead image credit",
    "detailsComponent.profiles": "Profiles",
    "detailsComponent.lead": "Lead",
    "detailsComponent.card": "Card",
    "detailsComponent.default": "Default",
    "detailsComponent.textForBodySection": "Text for body section",
    "listComponent.id": "Id",
    "listComponent.name": "Name",
    "headerComponent.home": "Home",
    "headerComponent.articles": "Articles"
}