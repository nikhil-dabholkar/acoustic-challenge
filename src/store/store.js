import rootReducer from '../reducer/index';
import thunk from 'redux-thunk';
import { createStore, applyMiddleware } from 'redux';

export const store = createStore(rootReducer, applyMiddleware(thunk))