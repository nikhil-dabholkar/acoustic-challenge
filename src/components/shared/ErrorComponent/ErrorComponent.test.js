import React from 'react';
import ErrorComponent from './ErrorComponent';
import RootContainer from '../../../containers/RootContainer/RootContainer';
import { IntlProvider } from 'react-intl';
import { store } from "../../../store/store";
import { Provider } from 'react-redux';
import { render, cleanup, screen } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import { Router } from 'react-router-dom'
import { createMemoryHistory } from 'history'
import en_US from "../../../i18n/en_US.json";
import App from '../../../App';
import TestRenderer from 'react-test-renderer'; // ES6

afterEach(cleanup);

function renderWithRouter(
    ui,
    { route = '/', history = createMemoryHistory({ initialEntries: [route] }) } = {}
) {
    return {
        ...render(<Router history={history}>{ui}</Router>),
        history,
    }
}

function rendererWithRouter(
    ui,
    { route = '/', history = createMemoryHistory({ initialEntries: [route] }) } = {}
) {
    return {
        ...renderer.create(<Router history={history}>{ui}</Router>),
        history,
    }
}

describe('ErrorComponent', () => {
    test('Check if ErrorComponent Renders', async () => {
        TestRenderer.act(() => {
            jest.useFakeTimers();
            const route = '/articles/invalid_route';
            const { container } = renderWithRouter(<Provider store={store}><IntlProvider locale="en" messages={en_US}><RootContainer></RootContainer></IntlProvider></Provider>, { route })
            setTimeout(async () => {
                console.log(container.innerHTML);
                const lazyElement = await screen.findByTestId("errorComponent")
                expect(lazyElement).toBeInTheDocument();
            }, 25);
            jest.runAllTimers();
        })

    })

    it("Check if matches ErrorComponent Snapshot", () => {
        const route = '/invalid_route';
        const errorObj = {
            response: {
                data: {
                    errors: {
                        code: 56009,
                        message: en_US["rootContainer.invalidURL"]
                    }
                }
            }
        }
        const tree = rendererWithRouter(<Provider store={store}><IntlProvider locale="en" messages={en_US}><ErrorComponent error={errorObj} ></ErrorComponent></IntlProvider></Provider>, { route }).toJSON();
        expect(tree).toMatchSnapshot();
    })
})
