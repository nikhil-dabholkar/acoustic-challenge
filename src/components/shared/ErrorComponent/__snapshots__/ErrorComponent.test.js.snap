// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorComponent Check if matches ErrorComponent Snapshot 1`] = `
<div
  className="container"
  data-testid="errorComponent"
>
  <div
    className="bx--inline-notification bx--inline-notification--error bx--inline-notification--hide-close-button"
    kind="error"
    role="alert"
  >
    <div
      className="bx--inline-notification__details"
    >
      <svg
        aria-hidden={true}
        className="bx--inline-notification__icon"
        fill="currentColor"
        focusable="false"
        height={20}
        preserveAspectRatio="xMidYMid meet"
        viewBox="0 0 20 20"
        width={20}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10,1c-5,0-9,4-9,9s4,9,9,9s9-4,9-9S15,1,10,1z M13.5,14.5l-8-8l1-1l8,8L13.5,14.5z"
        />
        <path
          d="M13.5,14.5l-8-8l1-1l8,8L13.5,14.5z"
          data-icon-path="inner-path"
          opacity="0"
        />
        <title>
          error icon
        </title>
      </svg>
      <div
        className="bx--inline-notification__text-wrapper"
      >
        <p
          className="bx--inline-notification__title"
        >
          56009: 
        </p>
        <div
          className="bx--inline-notification__subtitle"
        >
          <span>
            The requested URL is invalid please check the URL and retry.
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
